snudown.h

```diff
@@ -0,0 +1,118 @@
+/* snudown.h - header for snudown */
```

Comment:
As far as I remember, definition is ok, implementations is not the best practice. I checked other `h` files and they almost all have function definitions. But agree, if we have some one who has good experience in C, we can ask him\her